core/quality/sender.go

```diff
@@ -224,6 +247,43 @@ func (sender *Sender) sendEvent(eventName string, context interface{}) {
 	}
 }
 
+func (sender *Sender) rememberSessionContext(context sessionContext) {
+	sender.sessionsMu.Lock()
+	defer sender.sessionsMu.Unlock()
+
+	sender.sessionsActive[context.ID] = context
+}
+
+func (sender *Sender) forgetSessionContext(context sessionContext) {
+	sender.sessionsMu.Lock()
+	defer sender.sessionsMu.Unlock()
+
+	delete(sender.sessionsActive, context.ID)
+}
+
+func (sender *Sender) recoverSessionContext(sessionID string) (sessionContext, error) {
+	sender.sessionsMu.Lock()
```

Comment:
As sessionsMu is RWMutex I probably wanted RLock, RUnlock here?

Reply:
Yes